Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/filter/typefilters/PathFilter.java

```diff
@@ -299,14 +300,18 @@ private static boolean haveSameAttributeValue(ActivityInstance act1, ActivityIns
     private static boolean confirmInterval(OperationType operationType, ActivityInstance act1, ActivityInstance act2,
                                            double interval) {
 
+        CalendarModel calendarModel = act1.getCalendarModel();
+
         double duration = 0;
 
         if (act2 == null) {
             duration = act1.getDuration();
         } else {
             long st = act1.getStartTime();
             long et = act2.getEndTime();
-            duration = et > st ? et - st : 0;
+            duration = calendarModel != null ?
+                    Long.valueOf(calendarModel.getDuration(new long[]{st}, new long[]{et})[0]).doubleValue() :
```

Comment:
Can we add a utility method, e.g. CalendarDuration.getDuration(CalendarModel calendarModel), add the above code to that method, and then use this method everywhere? It is to the above code duplicated in many places, and some places may forget to do it correctly leading to errors.

Reply:
Can we use the simpler API provided in CalendarModel (below) rather than the above method with conversion to long array? 
- DurationModel getDuration(Long starDateTimeUnixTs, Long endDateTimeunixTs)
Can use DurationModel.getDuration.getNano() or DurationModel.getDuration.get(ChronoUnit.MILLIS).

Reply:
DurationModel.getDuration.getNano() not applicable
DurationModel.getDuration.get(ChronoUnit.MILLIS) caused error

Reply:
Pls use Duration.toMillis: https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toMillis--